composer/models/huggingface.py

```diff
@@ -302,6 +300,14 @@ def eval_forward(self, batch, outputs: Optional[Any] = None):
             # if we are in the single class case, then remove the classes dimension
             if output.shape[1] == 1:
                 output = output.squeeze(dim=1)
+        else:
+            output = outputs if outputs else self.forward(batch)
+
+        if self.shift_labels or batch.get('mode', None) == 'icl_task':
```

Comment:
can you just move this code back to where it was? you can still change the conditional to be this or statement with icl task